src/Serialization/Fields.jl

```diff
@@ -268,6 +268,79 @@ function load_internal_with_parent(s::DeserializerState,
     return parent(num, den)
 end
 
+################################################################################
+# RationalFunctionField
+
+@registerSerializationType(AbstractAlgebra.Generic.RationalFunctionField, "RationalFunctionField")
+
+function save_internal(s::SerializerState,
+                       RF::AbstractAlgebra.Generic.RationalFunctionField)
+    return Dict(
+        :base_ring => save_type_dispatch(s, base_ring(RF)),
+        :symbols => save_type_dispatch(s, symbols(RF))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: AbstractAlgebra.Generic.RationalFunctionField},
+                       dict::Dict)
+    R = load_unknown_type(s, dict[:base_ring])
+    symbols = load_unknown_type(s, dict[:symbols])
+
+    return RationalFunctionField(R, symbols, cached=false)[1]
+end
+
+#elements
+@registerSerializationType(AbstractAlgebra.Generic.Rat, "Rat")
+
+function save_internal(s::SerializerState, f::AbstractAlgebra.Generic.Rat)
+    frac_elem_parent = save_type_dispatch(s, parent(denominator(f)))
+    return Dict(
+        :parent => save_type_dispatch(s, parent(f)),
+        :frac_elem_parent => frac_elem_parent,
+        :den => save_type_dispatch(s, denominator(f)),
+        :num => save_type_dispatch(s, numerator(f))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: AbstractAlgebra.Generic.Rat},
+                       dict::Dict)
+    _ = load_unknown_type(s, dict[:frac_elem_parent])
+    R = load_type_dispatch(s, AbstractAlgebra.Generic.RationalFunctionField, dict[:parent])
+    num = load_unknown_type(s, dict[:num])
+    den = load_unknown_type(s, dict[:den])
```

Comment:
Based on what @tthsqe12 wrote, perhaps this would help?
```suggestion
    num = load_unknown_type(s, dict[:num]; parent= base_ring(R))
    den = load_unknown_type(s, dict[:den]; parent= base_ring(R))
```

Reply:
if `R` is `QQ(x,y)`, then `base_ring(R)` is just `QQ`. To get the `QQ[x,y]`, you have to use some non-exported functions.